### PR TITLE
#356 Diversity report shows data for each exercise stage

### PR DESCRIPTION
**Pull request by warrensearle**
- Diversity report shows data for all exercise stages.
- Data for each stage can be exported
- Data for all stages can be exported
- Print view tidied slightly

Example exercise on develop: exercises/wdpALbyICL7ZxxN5AQt8/reports/diversity

### CODE REVIEW

File: src/views/Exercises/Show.vue [hunk 1]
```diff
@@ -122,15 +122,19 @@ export default {
...
       ];
       if (this.exercise.applicationsCount || this.hasOpened) {
+        const children = [
+          { page: 'Draft', name: 'exercise-show-applications-in-status', params: { status: STATUS.DRAFT } },
+          { page: 'Applied', name: 'exercise-show-applications-in-status', params: { status: STATUS.APPLIED } },
+          { page: 'Withdrawn', name: 'exercise-show-applications-in-status', params: { status: STATUS.WITHDRAWN } },
+        ];
+        if (this.exercise.applicationRecords) {
+          children.push({ page: 'Independent Assessments', name: 'exercise-show-independent-assessments' });
+          // children.push({ page: 'Consent Forms', name: 'exercise-show-independent-assessments' });
```
Comment (lloback) [inline src/views/Exercises/Show.vue:132]: this is something that we are pushing before the 16th?

File: src/views/Exercises/Show/Reports/Diversity.vue [hunk 2]
```diff
@@ -25,8 +38,10 @@
               </button>
             </div>
           </div>
+          1
```
Comment (lloback) [inline src/views/Exercises/Show/Reports/Diversity.vue:41]: I didn't get what this number is doing here

**Comment by lloback** [comment]
I run this branch locally and this is a really good progress!

File: src/views/Exercises/Show.vue (hunk 1 above)
Comment (warrensearle) [reply src/views/Exercises/Show.vue:132]: It's a nice to have... but have removed the comment for now. Was just playing around with where to put the links in our massive side nav!
On that regard I've suggested that we move to horizontal tabs/menu for Overview | Applications | Stages | Reports and then the side navigation can focus on the links relevant to the current tab/section.

File: src/views/Exercises/Show/Reports/Diversity.vue (hunk 2 above)
Comment (warrensearle) [reply src/views/Exercises/Show/Reports/Diversity.vue:41]: Oops

**Comment by lloback** [comment]
Please merge whenever you will.

File: src/views/Exercises/Show.vue [hunk 3]
```diff
@@ -122,15 +122,18 @@ export default {
...
         pages.push({ 
           page: 'Applications',
           name: 'exercise-show-applications',
-          children: [
-            { page: 'Draft', name: 'exercise-show-applications-in-status', params: { status: STATUS.DRAFT } },
-            { page: 'Applied', name: 'exercise-show-applications-in-status', params: { status: STATUS.APPLIED } },
-            { page: 'Withdrawn', name: 'exercise-show-applications-in-status', params: { status: STATUS.WITHDRAWN } },
-            { page: 'Independent Assessments', name: 'exercise-show-independent-assessments' },
-          ],
+          children: children,
```
[the remark this answers is not in the document]
Comment (warrensearle) [reply src/views/Exercises/Show.vue:136]: 👍

**Pull request merged**